components/brave_vpn/common/wireguard/wireguard_utils.cc

```diff
@@ -48,6 +51,64 @@ constexpr char kWireguardConfigTemplate[] = R"(
   Endpoint = {vpn_server_hostname}:51821
 )";
 
+bool IsValidConfigDNS(const std::string& config_str) {
+  int match_count = 0, any_match_count = 0;
+
+  // Match our expected `DNS` field.
+  absl::string_view input(config_str);
+  while (match_count < 2 &&
+         re2::RE2::FindAndConsume(&input, R"((?m)^\s*DNS = 1\.1\.1\.1$)")) {
```

Comment:
seems a bit odd to me that we're using regex to validate things like ip address and hostnames since there are better tools for that in chromium. Wouldn't it be better to capture the field value and validate it using GURL, etc... ?

Reply:
I think with `GURL` you still have to add a schema. There`s also `net::HostPortPair`.

Reply:
With our code, we're hardcoding `1.1.1.1` as the DNS. Since that's the case, I think it's fine to do a text match like this. Unless we change the default, there's no need to validate (it's either right or wrong)